src/AutoRest.CSharp.V3/Output/Models/Types/ObjectType.cs

```diff
@@ -205,17 +208,21 @@ private ObjectTypeConstructor BuildInitializationConstructor()
                 }
                 else if (IsStruct || property.SchemaProperty?.IsRequired == true)
                 {
-                    // For structs all properties become required
-                    Constant? defaultParameterValue = null;
                     if (property.SchemaProperty?.ClientDefaultValue is object defaultValueObject)
                     {
                         defaultParameterValue = BuilderHelpers.ParseConstant(defaultValueObject, propertyType);
                     }
 
+                    var inputType = TypeFactory.GetInputType(propertyType);
+                    if (defaultParameterValue != null && TypeFactory.IsStruct(property.ValueType))
+                    {
+                        inputType = inputType.WithNullable(true);
```

Comment:
Do we need to change `defaultParameterValue` to null here?

Reply:
No, this is for the case where we make the parameter optional if the default value is given for the extensible enums. We would express it using a nullable parameter that falls back to the default value when null.

Reply:
I think you are mixing two things up here: parameter default value and initializer default value

```
                             v- Parameter default value
Constructor(int? parameter = 3)
{
	Property = parameter ?? 4;
}                           ^ - initializer default value
```

Currently you are setting them to the same thing and then trying to filter out in the writer using the conditions. I think it's more correct to set them right here. In the case you've mentioned parameter default value should be null and initializer default value should be the enum.

Reply:
I got what you are saying but [this code](https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp.V3/Generation/Writers/CodeWriterExtensions.cs#L38) checks the default value before setting the default value of ctor parameter. So, we cannot set defaultParameterValue to null here.

Reply:
We have a special constant value that represents null. `Constant.Default(type)`.